sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/UnivocityParser.scala

```diff
@@ -203,19 +203,11 @@ class UnivocityParser(
     }
   }
 
-  private val doParse = if (requiredSchema.nonEmpty) {
```

Comment:
This change looks good to me though, we need this in this pr? This fix should be addressed in #21909?

Reply:
Right, switching to 2.7.3 is valuable itself. I will revert the changes and postpone it to #21909 .